packages/modules-runtime-hot/installer.js

```diff
@@ -204,6 +204,16 @@ makeInstaller = function (options) {
   }
 
   function makeMissingError(id) {
+    const path = String(id)
+      .split('/');
+
+    const importsFrom = (location) =>
+      path.some((subPath) => subPath === location);
+
+    if (importsFrom('server') || importsFrom('client')) {
+      return new Error('Cannot import module ' + id + ' \n (cross-boundary import) \n see: https://guide.meteor.com/structure.html#special-directories');
```

Comment:
Would it be possible to make the error message more clear on why there is an error, and how to fix it? Maybe something like:

```
Unable to import on the client a module from a server directory: ${id}
```
```
Unable to import on the server a module from a client directory: ${id}
```